### PR TITLE
Pure component destructuring

### PR DESCRIPTION
**Pull request by andy-j-d**
Adds a ~~`destructure`~~ `destructuring` option to `pure-component` that will destructure the props where it's safe to do so. 
See the tests for use cases:
[Input](https://github.com/andy-j-d/react-codemod/blob/pure-component-destructuring/transforms/__testfixtures__/pure-component-destructure.input.js)
[Output](https://github.com/andy-j-d/react-codemod/blob/pure-component-destructuring/transforms/__testfixtures__/pure-component-destructure.output.js)

### CODE REVIEW

**Comment by cpojer** [comment]
@keyanzhang is the man to review this.

**Comment by keyanzhang** [comment]
Will do as soon as I find a stable internet connection!

--  
Keyan Zhang

File: transforms/pure-component.js [hunk 1]
```diff
@@ -85,31 +86,83 @@ module.exports = function(file, api, options) {
...
-        j.variableDeclarator(
-          j.identifier(name),
-          j.arrowFunctionExpression(
-            [buildIdentifierWithTypeAnnotation('props', typeAnnotation)],
-            body
-          )
-        ),
-      ]
-    );
+  const build = ({ functionType }) => (name, body, typeAnnotation, destructure) => {
```
Comment (keyanzhang) [inline transforms/pure-component.js]: nit: unnecessary to use a named parameter here

**Comment by slorber** [comment]
hey.

Just a note as I already passed my codemod: I would have liked to use this option, but also why not using an arrow function instead of replacing the class by a real function?

**Comment by andy-j-d** [comment]
@slorber if you use both the `destructure` (which will become `destructuring`) and `useArrows` options, you'll get destructured arrow functions.

**Comment by slorber** [comment]
so bad I already did the migration, the useArrow option was not documented :)

**Comment by andy-j-d** [comment]
I've put in those fixes @keyanzhang

**Comment by andy-j-d** [comment]
@keyanzhang :ping:

**Comment by keyanzhang** [comment]
Sorry for the late response -- somehow I missed the email notification. Thanks for the great work @andy-j-d!

**Pull request merged**